packages/components/src/tooltip.js

```diff
@@ -8,42 +8,50 @@ export class Tooltip extends ArticleElement {
     this.visible = false;
     this.addEventListener('keydown', this.keyDown);
     this.addEventListener('mousedown', this.mouseDown);
-    this.addEventListener('focusout', this.focusOut);
   }
 
-  mouseDown(event) {
-    if (this.visible) {
-      this.hide();
-    } else {
-      this.show();
-    }
-  }
+  keyDown(event) {
+    if (event.key !== 'Enter' || this.visible) return;
 
-  focusOut(event) {
-    if (!this.contains(event.relatedTarget)) {
-      this.hide();
-    }
+    this.show();
   }
 
-  keyDown(event) {
-    if (event.key == 'Enter') {
-      this.show();
-    } else if (event.key == 'Escape') {
-      this.hide();
-    }
+  mouseDown() {
+    if (this.visible) return;
+
+    this.show();
   }
 
   hide() {
     this.querySelector('.tooltip').style.display = 'none';
     this.visible = false;
+
+    // clean up event listeners
+    document.removeEventListener('keydown', this.keyDownClose);
+    document.removeEventListener('mousedown', this.mouseDownClose);
   }
 
   show() {
     const bbox = this.getBoundingClientRect();
     const ttip = this.querySelector('.tooltip');
-    ttip.style.transform = `translate(-${bbox.width}px, ${4 + bbox.height}px)`;
+    ttip.style.transform = `translate(${bbox.left - bbox.width < 0 ? 0 : -bbox.width}px, calc(1em + 6px))`;
```

Comment:
What is the rationale for removing `bbox.height` here? What problem is being solved?

Reply:
When the tooltip body wraps lines, the border box height for the body ends up doubling (if it wraps two lines instead of one). Since the tooltip transform is calculated from the end of the tooltip body, it leaves a gap equal to half the border box height from the tooltip body to the tooltip. 

Possible solutions:
- Move the tooltip element before the tooltip body (committed this solution)
  - This would let the border box height work regardless of body height
- If the tooltip body is expected to be at most two lines after text wrapping, shift it half or the full height of the border box depending on the case
- Use a fixed number
  - This would ensure that the tooltip is at the same y position relative to the body